modules/ROOT/pages/indexes-for-nearest-neighbor.adoc

```diff
@@ -0,0 +1,436 @@
+A common semantic of a vector is as an embedding, a numerical representation of a resource, for example, text, images, audio, or documents.
+Embedding generators such as https://cloud.google.com/vertex-ai[Vertex AI] or https://openai.com/[OpenAI] can produce 768, 1024, 1536, etc., dimensional vector embeddings.
+
+== Vector index commands and procedures
+
+Vector indexes are managed through Cypher commands and built-in procedures, see link:{neo4j-docs-base-uri}/operations-manual/{page-version}/reference/procedures[Operations Manual -> Procedures] for a complete reference.
+
+The procedures and commands for vector indexes are listed in the following table:
+
+.Commands and procedures for vector indexes
+[options="header", cols="3,4a,5a"]
+|===
+| Usage | Procedure/Command | Description
+
+| Create vector index.
+| {link-procedures-reference}#procedure_db_index_vector_createNodeIndex[`db.index.vector.createNodeIndex`]
+| Create a named vector index for the given label and property with given vector dimensionality using the given similarity function.
+
+| Use vector index.
+| {link-procedures-reference}#procedure_db_index_vector_queryNodes[`db.index.vector.queryNodes`]
+| Query the given vector index. Returns the requested number of approximate nearest neighbor nodes and their similarity score, ordered by score.
+
+| Drop vector index.
+| `+DROP INDEX index_name+`
+| Drop the specified index.
+
+| Listing all vector indexes.
+| `SHOW INDEXES WHERE type = "VECTOR"`
+| Lists all vector indexes, see the xref:indexes-for-search-performance.adoc#indexes-list-indexes[`SHOW INDEXES`] command for details. There is no vector index filter built into `SHOW INDEXES`.
+
+| Set vector property.
+| {link-procedures-reference}#procedure_db_create_setVectorProperty[`db.create.setVectorProperty`]
+| Update a given node property with the given vector in a more space-efficient way than directly using xref:clauses/set.adoc#set-set-a-property[`SET`].
+
+|===
+
+[[indexes-vector-create]]
+== Create and configure vector indexes
+
+You can create vector indexes using the procedure {link-procedures-reference}#procedure_db_index_vector_createNodeIndex[`db.index.vector.createNodeIndex`].
+The name of the index must be unique so that you can reference it when xref:#indexes-vector-query[querying] or if you want to xref:indexes-vector-drop[drop] the index.
+
+[IMPORTANT]
+====
+The index name must be unique among both indexes and constraints.
+====
+
+A vector index is a single-label, single-property index for nodes.
+In addition to the label and property key, both given as `STRING`, a vector index needs to be configured with both the dimensionality of the vector, `INTEGER`, and the measure of similarity between two vectors, case-insensitive `STRING` For details, see xref:#indexes-vector-similarity[].
+
+.Syntax for creating vector indexes
+[source,cypher]
+----
+CALL db.index.vector.createNodeIndex('index_name', 'LabelName', 'propertyKey', dimensionality, 'similarity_function')
```

Comment:
The test for this code is failing because it thinks `dimensionality` is a variable.

Reply:
… ok, how interesting … does the test try and execute cypher?
eitherway I changed this to just be the signature, as was done by you later on.

Reply:
@parnmatt yes, see here https://github.com/neo4j/docs-testing/tree/main/test-cypher
TL;DR: don't mix `[source, cypher]` with `[source, syntax]`, as all cypher blocks get executed for testing.